frontend/apps/reader/modules/readertoc.lua

```diff
@@ -210,6 +212,24 @@ function ReaderToc:validateAndFixToc()
         local seq = (cur_seq_by_level[depth] or 0) + 1
         item.seq_in_level = seq
         cur_seq_by_level[depth] = seq
+        -- Also use this loop to compute chapter lengths
```

Comment:
It's probably a bit premature optimization type of thinking, but that comes across as a lot of effort to do when someone doesn't have that setting enabled. :-)

Reply:
You're right - I initially stuck it here to avoid another loop, before I realized I needed other small loops inside...
But as I had to duplicate it at the bottom of the function for the "bogus toc" case, I can just externalize it to another method - and call it only when needed. Will do that.